bioblend/galaxy/objects/wrappers.py

```diff
@@ -1079,6 +1086,38 @@ def __init__(self, f_dict, container, gi=None):
         object.__setattr__(self, 'container', container)
 
     @property
+    def parent(self):
+        """
+        Get folder indicated by 'parent_id'
+        The root folder will have the library as a parent.
+        """
+        if self._cached_parent is None:
+            object.__setattr__(self, 
+                               '_cached_parent', 
+                               self._get_parent())
+        return self._cached_parent
+
+    def _get_parent(self):
+        """
+        Return folder indicated by 'parent_id'
+        """
+        parent_id = self.wrapped.get('parent_id',None)
+        if parent_id is not None:
+            if parent_id[0]!='F':
+                # older Galaxy versions strip the F
+                parent_id = u'F%s' % (parent_id)
+            try:
+                return self.container.get_folder(parent_id)
+            except bioblend.galaxy.client.ConnectionError:
+                # Depending on version, galaxy is returning a dummy parent_id 
+                #  for the root Folder
+                # Clear the parent_id, so we don't try to load it each time
+                self.wrapped['dummy_parent_id']=self.wrapped['parent_id']
+                self.wrapped['parent_id']=None
```

Comment:
Sorry, I don't do pep8 by habit yet. I have the checker installed. I'll
work through the errors.

On Wed, Feb 25, 2015 at 4:53 PM, Nicola Soranzo <notifications@github.com>
wrote:

> In bioblend/galaxy/objects/wrappers.py
> <https://github.com/galaxyproject/bioblend/pull/115#discussion_r25401236>:
>
> > +        """
> > +        Return folder indicated by 'parent_id'
> > +        """
> > +        parent_id = self.wrapped.get('parent_id',None)
> > +        if parent_id is not None:
> > +            if parent_id[0]!='F':
> > +                # older Galaxy versions strip the F
> > +                parent_id = u'F%s' % (parent_id)
> > +            try:
> > +                return self.container.get_folder(parent_id)
> > +            except bioblend.galaxy.client.ConnectionError:
> > +                # Depending on version, galaxy is returning a dummy parent_id
> > +                #  for the root Folder
> > +                # Clear the parent_id, so we don't try to load it each time
> > +                self.wrapped['dummy_parent_id']=self.wrapped['parent_id']
> > +                self.wrapped['parent_id']=None
>
> PEP8: missing spaces around "=" in the 2 previous lines.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/galaxyproject/bioblend/pull/115/files#r25401236>.
>